pkg/s3signer/request-signature-v2.go

```diff
@@ -42,9 +42,7 @@ const (
 func encodeURL2Path(u *url.URL) (path string) {
 	// Encode URL path.
 	if isS3, _ := filepath.Match("*.s3*.amazonaws.com", u.Host); isS3 {
```

Comment:
Okay :+1:

I am not sure exactly why mc is only accepting s3.amazonaws.com address format. But since this is low level, I will agree